src/utils/supertype.rs

```diff
@@ -186,6 +187,15 @@ pub fn get_supertype(l: &DataType, r: &DataType) -> Option<DataType> {
             //TODO(sammy): add time, struct related dtypes
             (Boolean, Float32) => Some(Float32),
             (Boolean, Float64) => Some(Float64),
+            // TODO(Clark): Add support for getting supertype for two fixed size lists once Arrow2 supports such a cast.
+            // (FixedSizeList(inner_left_field, inner_left_size), FixedSizeList(inner_right_field, inner_right_size)) if inner_left_size == inner_right_size => {
+            //     let inner_st = inner(&inner_left_field.dtype, &inner_right_field.dtype)?;
+            //     Some(DataType::FixedSizeList(Box::new(Field::new(inner_left_field.name.clone(), inner_st)), *inner_left_size))
+            // }
+            (FixedSizeList(inner_left_field, inner_left_size), List(inner_right_field)) => {
```

Comment:
Did you mean the right side to be `FixedSizeList` as well since `FixedSizeList` + `FixedSizeList` yields `FixedSizeList` if the length is the same. and `FixedSizeList` + `List` should give `List` as the supertype

Reply:
Casting between `FixedSizeListArray`s is not supported (see the commented out match arm + comment right above this), so this is intended; however, you are correct that the result type should be a `List()`, let me make that change!